### PR TITLE
Allow using v2 API to load forms

### PR DESCRIPTION
**Pull request by lfdebrux**
### What problem does this pull request solve?

Trello card: https://trello.com/c/OOYCWoTa/1779-8-change-forms-runner-to-use-v2-api 

This PR makes it so that if we enable the `api_v2` feature flag forms-runner uses the new v2 API.

Change `Api::V1::FormSnapshotRepository` to use new `Api::V2::FormDocumentResource` and `Api::V1::Converter` classes to allow this app to retrieve forms using the v2 API without changes to the rest of the codebase. Basically, we get the v2 form documents and then immediately convert them to the older form snapshot format so we can keep using the same models.

This behaviour is currently behind a feature flag and not enabled, so it should be safe to deploy this change to production.

I have tested the change locally with the feature flag enabled and the end to end tests pass, however the unit tests need more work to pass, which I will do in a future PR.

### Things to consider when reviewing

- Ensure that you consider the wider context.
- Does it work when run on your machine?
- Is it clear what the code is doing?
- Do the commit messages explain why the changes were made?
- Are there all the unit tests needed?
- Has all relevant documentation been updated?

### CODE REVIEW

File: config/settings/development.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+features:
+  api_v2: true
```
Comment (thomasiles) [inline config/settings/development.yml:2]: I think this is a typo - it should be `v2_api`.

Comment (lfdebrux) [reply config/settings/development.yml:2]: Good spot, thank you

Comment (lfdebrux) [reply config/settings/development.yml:2]: Actually I might change the feature flag to be properly named `api_v2`, so it's consistent with other places in our codebase...

Comment (lfdebrux) [reply config/settings/development.yml:2]: I went ahead and changed it, the feature flag should now be `api_v2` everywhere

**Review by thomasiles — APPROVED** [review]
Sorry I meant to leave the rest of the review beyond the typo yesterday but I got distracted.

I really like this approach, it's very clever! It really reduces the risks and allows us to migrate the runner incrementally.

My only comment was going to be that some sort of end to tests to cover the whole new journey would be useful as a sanity check that nothing has broken.

I've seen you already have another PR which covers it.

I tested it yesterday without the changes to the name of the feature flag and didn't find any issues.

Amazing 🏅

File: app/services/api/v1/converter.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+class Api::V1::Converter
+  def to_api_v1_form_snapshot(form_document)
```
Comment (chao-xian) [inline app/services/api/v1/converter.rb:2]: just a style note here: the `to_api_` prefix is superfluous as the inference that this is the `API::V1::Converter` class so a method that gives `form_snapshot(form_document)` would have been more succinct - similar for `page(step)` below. Or for a bit of clarity `v1_form_snapshot` would be ok too. Nbd as I understand this class wont be around too long.

**Pull request merged**